cli_tools_tests/e2e/gce_windows_upgrade/test_suites/windows_upgrade/windows_upgrade_test_suite.go

```diff
@@ -87,6 +90,20 @@ func TestSuite(
 		testsMap[testType][testBYOL] = runWindowsUpgradeBYOL
 	}
 
+	// These tests only apply to wrapper since it's testing staging install media.
+	stagingUpgrade2008To2012 := junitxml.NewTestCase(
+		testSuiteName, fmt.Sprintf("[%v] Staging test 2008 -> 2012", e2e.Wrapper))
+	stagingUpgrade2012To2016 := junitxml.NewTestCase(
```

Comment:
Sorry, if I wasn't clear in the last comment. I thought your previous entries were correct for
2012-> 2016
2012-> 2019
2012-> 2022

Just that we also needed starting upgrades from 2016 & 2019.

So in total, I believe we need:
2008->2012 (currently present)
2012->2016 (currently present)
2012->2019
2012->2022
2016->2019 (currently present)
2016->2022
2019->2022 (currently present)

Reply:
to be honest I don't think it's required to cover all the combinations, I want to keep the e2e test set simple and critical. However I am ok to add them since the previous import/export tests has been moved out and thus we have more test VM resources now.